cms-contentful/it/com/commercetools/sunrise/cms/contentful/ContentfulCmsServiceIT.java

```diff
@@ -52,65 +46,221 @@ public void whenCouldNotFetchEntry_thenThrowException() throws Exception {
+        ContentfulCmsService contentfulCmsService = ContentfulCmsService.of(spaceId(), token(), "finn", PAGE_TYPE_ID_FIELD_NAME);
+
+        Optional<CmsPage> content = contentfulCmsService.page("finn", emptyList()).toCompletableFuture().get(5, TimeUnit.SECONDS);
+
+        assertThat(content).isPresent();
+        Optional<String> field = content.get().field("mediaOneFile");
+        assertThat(field).isPresent();
+        assertThat(field.get()).isEqualTo("//images.contentful.com/l6chdlzlf8jn/2m1NzbeXYUksOGIwceEy0U/4e3cc53a96313a4bd822777af78a3b4d/some-5.jpg");
+    }
+
+    @Test
+    public void whenAskForContentInMediaArray_thenGetUrlElements() throws Exception {
+        ContentfulCmsService contentfulCmsService = ContentfulCmsService.of(spaceId(), token(), "finn", PAGE_TYPE_ID_FIELD_NAME);
+
+        Optional<CmsPage> content = contentfulCmsService.page("finn", emptyList()).toCompletableFuture().get(5, TimeUnit.SECONDS);
+
+        assertThat(content).isPresent();
+        Optional<String> field = content.get().field("mediaManyFiles[0]");
+        assertThat(field).isPresent();
+        assertThat(field.get()).isEqualTo("//images.contentful.com/l6chdlzlf8jn/6j9p38phC0oU0g42aqUSc4/2eb0c261bc13353ed867b13076af6b1f/logo.gif");
+
+        field = content.get().field("mediaManyFiles[1]");
+        assertThat(field).isPresent();
+        assertThat(field.get()).isEqualTo("//images.contentful.com/l6chdlzlf8jn/27BPx56xMcuGKe8uk8Auss/335581cd9daf3e9d0de254313e36d43b/some-5.jpg");
+    }
+
+    @Test
+    public void whenAskForContentInArrayOutsideTheScope_thenGet() throws Exception {
+        ContentfulCmsService contentfulCmsService = ContentfulCmsService.of(spaceId(), token(), "finn", PAGE_TYPE_ID_FIELD_NAME);
+        Optional<CmsPage> content = contentfulCmsService.page("finn", emptyList()).toCompletableFuture().get(5, TimeUnit.SECONDS);
+
+        assertThat(content).isPresent();
+        Optional<String> field = content.get().field("array[3].textArrayField");
+        assertThat(field).isNotPresent();
+    }
+
+    @Test
+    public void whenAskForContentWithGermanOrEmptyLocales_thenGetGermanTranslation() throws Exception {
+        ContentfulCmsService contentfulCmsService = ContentfulCmsService.of(spaceId(), token(), "finn", PAGE_TYPE_ID_FIELD_NAME);
+
+        Optional<CmsPage> content = contentfulCmsService.page("finn", SUPPORTED_LOCALES).toCompletableFuture().get(5, TimeUnit.SECONDS);
+
+        assertThat(content).isPresent();
+        Optional<String> field = content.get().field("array[3].textArrayField[1]");
+        assertThat(field).isPresent();
+        assertThat(field.get()).isEqualTo("zwei");
+
+        content = contentfulCmsService.page("finn", emptyList()).toCompletableFuture().get(5, TimeUnit.SECONDS);
+
+        assertThat(content).isPresent();
+        field = content.get().field("array[3].textArrayField[1]");
+        assertThat(field).isPresent();
+        assertThat(field.get()).isEqualTo("zwei");
```

Comment:
Right, that's a complex issue but I only added this test to verify behavior that was already implemented. So I'd suggest to treat it as a different issue. And it requires thorough concept.

In general it depends on the external requirements of this cms service API. I'd leave like it is now and if other cms systems support are required in the future then make it more generic and coherent.

My general opinion is that making it very generic in design is almost impossible ;)
If we wanted to cover all possible CMS providers APIs we'd end up with sth like
[Content_repository_API_for_Java](https://en.wikipedia.org/wiki/Content_repository_API_for_Java)
There was entire team of experts working on this JSR for a long time and it's quite sophisticated ;)